packages/lit-dev-content/site/tutorials/content/tooltip/07.md

```diff
@@ -0,0 +1,58 @@
+You've polished the tooltip's behavior with animations. Now it's time to ensure
+great performance. Since a tooltip displays only when the user interacts with its
+target, it doesn't need to be created until then. If a page uses a lot of
+tooltips, which is not uncommon, adding this optimization can speed up initial
+page rendering by a meaningful amount.
+
+To do this, you'll create a function users will call with a tooltip target
+element and a callback that will render tooltip content on demand when
+its needed. Add a `static` method to the `SimpleTooltip` class called
+`lazy`. It will setup event listeners on the target element that listen on the
+same events used to trigger the tooltip. The listener creates the tooltip
+element and calls the user callback with it to fill in the content before
+showing it. Near the top of the class definition, add:
+
+```ts
+static lazy(target: Element, callback: (target: SimpleTooltip) => void) {
+  let called = false;
+  enterEvents.forEach(name => target.addEventListener(name, () => {
+    if (!called) {
+      called = true;
+      const tooltip = document.createElement('simple-tooltip') as SimpleTooltip;
```

Comment:
can skip this assertion if you add the TS global HTMLTagNameMap declaration at the bottom

Reply:
Done